meson.build

```diff
@@ -0,0 +1,90 @@
+project(
+  'xdg-dbus-proxy',
+  'c',
+  version : '0.1.3',
+  meson_version : '>=0.49.0',
+  default_options : [
+    'warning_level=2',
+  ],
+)
+
+cc = meson.get_compiler('c')
+add_project_arguments('-D_GNU_SOURCE', language : 'c')
+common_include_directories = include_directories('.')
+
+add_project_arguments(
+  cc.get_supported_arguments([
+    # Deliberately not warning about this, ability to zero-initialize
+    # a struct is a feature
+    '-Wno-missing-field-initializers',
+    '-Wno-error=missing-field-initializers',
+
+    # Deliberately not warning about this, unused parameters to a callback
+    # are not a bug
+    '-Wno-unused-parameter',
+    '-Wno-error=unused-parameter',
+  ]),
+  language : 'c',
+)
+
+glib_dep = dependency('glib-2.0', version : '>=2.40')
+gio_dep = dependency('gio-2.0', version : '>=2.40')
+gio_unix_dep = dependency('gio-unix-2.0', version : '>=2.40')
+common_deps = [glib_dep, gio_dep, gio_unix_dep]
+
+if meson.is_subproject()
+  bindir = get_option('libexecdir')
+
+  if get_option('program_prefix') == ''
+    error('program_prefix option must be set when xdg-dbus-proxy is a subproject')
+  endif
+
+  exe_name = get_option('program_prefix') + 'dbus-proxy'
+else
+  bindir = get_option('bindir')
+  exe_name = 'xdg-dbus-proxy'
+endif
+
+conf_data = configuration_data()
+conf_data.set_quoted('BINDIR', get_option('prefix') / bindir)
```

Comment:
```suggestion
conf_data.set_quoted('BINDIR', join_paths(get_option('prefix'), bindir))
```

Reply:
Is there a reason to prefer the more verbose form? I was under the impression that the `/` operator (introduced in 0.49.0, which happens to be the minimal version I'm targeting here) is preferred.

Reply:
> (since 0.49.0) Using the / operator on strings is equivalent to calling [join_paths()](https://mesonbuild.com/Reference-manual_functions.html#join_paths).
>
> — https://mesonbuild.com/Reference-manual_functions.html#join_paths

Reply:
Okay, I would've thought the function would be more portable, but looks like I'm wrong.

Reply:
The function is *slightly* more portable to very old versions, but I'm using other features of 0.49+ anyway, so avoiding the `/` operator wouldn't gain us anything.